near-sdk/src/json_types/hash.rs

```diff
@@ -1,6 +1,10 @@
 use crate::CryptoHash;
 use borsh::{BorshDeserialize, BorshSerialize};
-use std::convert::{TryFrom, TryInto};
+use serde::{de, ser, Deserialize};
+use std::borrow::Cow;
```

Comment:
We should try to rebuild some contract that uses `Base58CryptoHash` with the contract builder, to compare the compilation size with using `Cow`. I don't expect anything dramatic, but just to be careful

Reply:
if you give me a workflow or point me in the direction of one that does, I can do that. I'd be curious to see if it actually increases the compiled code (and to what scale).

Reply:
Not sure what's the existing workflow, but I'd suggest adding a smoke test for this.

Something stupid like:

```rust
#[test]
fn test_we_are_smol() {
    std::process::Command::new("cargo")
      .current_dir("examples/locakable-fungible-token")
      .env("RUSTFLAGS", "--link-args=-s")
      .args(&["build", "--release", "--target", "wasm"])
      .status().unwrap()
    let wasm = std::fs::read("examples/locakable-fungible-token/target/wasm").unwrap();
    assert!(100k <= wasm && wasm <= 300k)
}
```

Reply:
You can use Base59CryptoHash in `examples/test-contract` and compile it before the change and after the change using contract builder https://github.com/near/near-sdk-rs/tree/master/contact-builder and compare binary sizes